configure.ac

```diff
@@ -927,6 +929,10 @@ AS_IF([test x"$enable_native_compiler" = "xno"],
   AC_MSG_NOTICE([the native compiler is disabled])],
   [native_compiler=true])
 
+AS_IF([$native_compiler],
+  [default_make_target=world.opt],
+  [default_make_target=world])
```

Comment:
In the equivalent option in https://github.com/ocaml/ocaml/pull/8835#issuecomment-520249008, you request that the new variables be `true` or `false` (which I'm actually just doing, which is why I noticed).

In this case, it would remove the need for `$default_make_target` completely, as you could do an `ifeq` based solely on `$(NATIVE_COMPILER)` in `Makefile`